lib/phlex/rails/form.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Phlex
+	module Rails
+		module Form
+			def initialize(model)
+				@model = model
+			end
+
+			def self.input_field(method_name, type:)
+				define_method method_name do |field, value: @model.attributes[field.to_s], **attributes|
+					input name: field_name(field),
+						type: type, value: value, **attributes
+				end
+			end
+
+			def template(&block)
+				form action: @url, method: @method do
+					authenticity_token_field
+					yield_content(&block)
+				end
+			end
+
+			def authenticity_token_field
+				input name: "authenticity_token",
+					type: "hidden", value: @_view_context.form_authenticity_token
+			end
+
+			def submit(value)
+				input name: "commit", type: "submit", value: value
+			end
+
+			def url
+				@_view_context.url_for(@model)
+			end
+
+			def field_name(field)
+				@_view_context.field_name(ActiveModel::Naming.param_key(@model.class), field)
+			end
```

Comment:
```suggestion
			def field_name(*field)
				@_view_context.field_name(ActiveModel::Naming.param_key(@model.class), *field)
			end
```

Allows `field_name(:its, :ya, index, :boy) => form[its][ya][0][boy]`